railgun/internal/ctrlutils/utils.go

```diff
@@ -141,3 +146,31 @@ func CRDExists(client client.Client, gvr schema.GroupVersionResource) bool {
 	}
 	return true
 }
+
+// retrieve Kong Admin API URL from configured name/namespace service
+func RetrieveKongAdminAPIURL(ctx context.Context, KongAdminAPI string, kubeCfg *rest.Config) (string, error) {
+	namespace, name, err := util.ParseNameNS(KongAdminAPI)
+	if err != nil {
+		return "", fmt.Errorf("failed to pars kong admin api namespace and name. err %v", err)
+	}
+	CoreClient, _ := clientset.NewForConfig(kubeCfg)
+	svc, err := CoreClient.CoreV1().Services(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to creating kubernetes client")
+	}
+	ingresses := svc.Status.LoadBalancer.Ingress
+	adminIP := ""
+	for _, ingress := range ingresses {
+		adminIP = ingress.IP
+	}
+
+	ports := svc.Spec.Ports
+	var adminPort int32
+	for _, port := range ports {
+		if port.Name == "kong-admin" {
+			adminPort = port.Port
+		}
+	}
```

Comment:
This is another place some light validation could improve the functionality of this code, I suggest we validate the `Spec.Ports` here to make sure it's in order and throw an error if not, so we don't end up with adminPort being set to the default int value of 0, which could break things in other places.

Reply:
I dont believe user is able to set a k8s resource port( targetport) as 0, that is against k8s rule.

Reply:
Correct, the _user_ could not. But in this case, it's the _code_ which can try to set it to `0` right? If we don't validate there's something there, the value will be initialized as 0 as this is the default for the `int32` var `adminPort`?